Add VC Presentation for Native Wallet docs. (#41)
credential-handler/chapi.io: developers/wallets/native/index.md

```

See the [OID4VCI Specification](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html) for more details on this protocol.

## Verifiable Credential Presentation

Browser side verification flows will open the CHAPI wallet selector in the user's
browser. If they select a native wallet, that URL will open the pre-registered
native wallet apps link (see above for how to register...).

That URL will to the Wallet's pre-registered domain--which will (thanks to the
magic of app links!) open in the user's wallet:
```
https://wallet.example.com/switchboard?request=
```

Depending on the protocols supported, the value of request will contain one
or more encoded JSON objects with nifty details...

...explain when this happens...maybe with mermaid chart?

### VC-API

The `vcapi` property within the object parsed above will contain a URL the wallet can use to handle that request.

> To respond with one or more credentials via a VC-API exchanger, the wallet MUST first send a `POST` request

Reviewer: And sometimes no credentials at all! (DID Authn only).

Author: @dlongley How about "zero or more"?

Reviewer: "zero or more" works for me.

to the URL (extracted from `protocols.vcapi` above) with an empty JSON object

> (`{}`) in the body of the request to attempt to download the VC to be stored.

Reviewer: ```suggestion
(`{}`) in the body of the request to attempt to get a verifiable presentation request with
the details about what the verifier wants.
```

```http
POST /exchangers/z1A1GqykGBWKbwhFCDqFjMfnG/exchanges/z19mxa763DAKX7diL51kBFecZ
Host: exchanger.example.com
Content-Type: application/json

{}
```

The response to that request will be a `verifiablePresentationRequest`
object. The `verifiablePresentationRequest` object will describe
what is required to continue the exchange.

Below is an example `verifiablePresentationRequest` containing a `QueryByExample`
query:

```json
{
  "verifiablePresentationRequest": {
    "query": [
      {
        "type": "QueryByExample",
        "credentialQuery": {
          "reason": "Please present your Verifiable Credential to complete the verification process.",
          "example": {
            "@context": [
              "https://www.w3.org/2018/credentials/v1",
              "https://www.w3.org/2018/credentials/examples/v1"
            ],
            "type": [
              "UniversityDegreeCredential"
            ]
          }
        }
      }
    ]
  }
}
```

> The wallet should fullfil the request by sending another `POST` request to the

Reviewer: ```suggestion
The wallet should fulfil the request by sending another `POST` request to the
```

Author: Almost...`fulfill` 😀

```

See the [OID4VCI Specification](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html) for more details on this protocol.

## Verifiable Credential Presentation

Browser side verification flows will open the CHAPI wallet selector in the user's
browser. If they select a native wallet, that URL will open the pre-registered
native wallet apps link (see above for how to register...).

That URL will to the Wallet's pre-registered domain--which will (thanks to the
magic of app links!) open in the user's wallet:
```
https://wallet.example.com/switchboard?request=
```

Depending on the protocols supported, the value of request will contain one
or more encoded JSON objects with the applicable details needed for using that
protocol.

{% mermaid %}
sequenceDiagram
  # User visits a Verifier's website and interacts with it to trigger a verification

  participant site as Verifier Site
  participant chapi as User's Browser / CHAPI
  participant app as Native Wallet App
  participant exchanger as VC-API Exchanger

  site ->> exchanger: 1. creates an exchange within a workflow
  exchanger ->> site: 2. JSON response contains the exchange URL/ID

>   site ->> chapi: 3. triggers CHAPI `get()` request w/blank VP + `protocols.vcapi` containing URL

Reviewer 2: ```suggestion
  site ->> chapi: 3. triggers CHAPI `get()` request w/blank `VerifiablePresentation` object + `protocols.vcapi` containing URL
```

Author: @brianorwhatever we're going to need code examples for this also, 'cause it's hard to even guess at what a "blank `VerifiablePresentation` object might be...but I'll commit the text change.
